File: C++(3)/7.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cout << "Ingrese la cantidad de elementos: ";
    cin >> n;

    vector<int> vector(n);
    for (int i = 0; i < n; i++) {
        cout << "Elemento " << i+1 << ": ";
        cin >> vector[i];
    }

    // Insertion Sort
    for (int i = 1; i < n; i++) {
        int clave = vector[i];
        int j = i - 1;
        while (j >= 0 && vector[j] > clave) {
            vector[j+1] = vector[j];
            j--;
        }
        vector[j+1] = clave;
    }

    cout << "Vector ordenado: ";
    for (int num : vector) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: Py (3)/7.py
def insercion_vector(vector):
    for i in range(1, len(vector)):
        clave = vector[i]
        j = i - 1
        while j >= 0 and vector[j] > clave:
            vector[j+1] = vector[j]
            j -= 1
        vector[j+1] = clave
    return vector

# Programa principal
n = int(input("Ingrese la cantidad de elementos: "))
vector = []
for i in range(n):
    valor = int(input(f"Elemento {i+1}: "))
    vector.append(valor)

print("Vector ordenado:", insercion_vector(vector))
